packages/components/pie-notification/test/visual/pie-notification.spec.ts

```diff
@@ -1,14 +1,273 @@
+
+test.beforeEach(async ({ mount }, testInfo) => {
+    testInfo.setTimeout(testInfo.timeout + 40000);
+
+    // This ensures the tag and icon components are registered in the DOM for each test.
+    // It appears to add them to a Playwright cache which we understand is required for the tests to work correctly.
+    const tagComponent = await mount(PieNotification);
+    await tagComponent.unmount();
+    const iconComponent = await mount(IconHeartFilled);
+    await iconComponent.unmount();
+});
+
+componentVariants.forEach((variant) => test(`should render all prop variations for Variant: ${variant}`, async ({ page, mount }) => {
+    await Promise.all(componentPropsMatrixByVariant[variant].map(async (combo: WebComponentPropValues) => {
+        const testComponent: WebComponentTestInput = createTestWebComponent(combo, renderTestPieNotification);
 
-test.describe('PieNotification - Visual tests`', () => {
-    test('should display the PieNotification component successfully', async ({ page, mount }) => {
-        await mount(PieNotification, {
-            props: {} as NotificationProps,
+        const propKeyValues = `
+            variant: ${testComponent.propValues.variant},
+            isCompact: ${testComponent.propValues.isCompact},
+            isDismissible: ${testComponent.propValues.isDismissible},
+            hideIcon: ${testComponent.propValues.hideIcon},
+            heading: ${testComponent.propValues.heading}
+        `;
+
+        const darkMode = ['neutral-alternative'].includes(variant);
+
+        await mount(
+            WebComponentTestWrapper,
+            {
+                props: { propKeyValues, darkMode },
+                slots: {
+                    component: testComponent.renderedString.trim(),
+                },
+            },
+        );
+    }));
+
+    // Follow up to remove in Jan
+    await page.waitForTimeout(5000);
+
+    await percySnapshot(page, `PIE Notification - Variant: ${variant}`, screenWidths);
+}));
+
+const initialValues: NotificationProps = {
+    isOpen: true,
+    isDismissible: true,
+    heading: 'Pie Notification Title',
+    leadingAction: {
+        text: 'Confirm',
+        ariaLabel: 'Button that confirm the action',
+        onClick: () => { console.info('confirmAction'); },
+    },
+};
+
+test.describe('Props', () => {
+    const confirmAction = () => { console.info('confirmAction'); };
+    const cancelAction = () => { console.info('cancelAction'); };
+    const mainAction = {
+        text: 'Confirm',
+        ariaLabel: 'Button that confirm the action',
+        onClick: confirmAction,
+    };
+    const secondaryAction = {
+        text: 'Cancel',
+        ariaLabel: 'Button that cancel the action',
+        onClick: cancelAction,
+    };
+
+    test.describe('PieNotification headingLevels', () => {
+        headingLevels.forEach((headingLevel) => {
+            test(`should render correctly the ${headingLevel} headingLevel`, async ({ page, mount }) => {
```

Comment:
i think this test can be combined with the main first test.
do you think we can pass headingLevel: headingLevels to line 28?

Reply:
@raoufswe I am not sure. Percy cannot handle several screenshots at once, by adding this prop we'll generate a lot of screenshots that looks the same due the combination of other props together.

Reply:
aha makes sense